src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,56 @@
+import { AriaSwitchProps, useSwitch } from '@react-aria/switch';
+
+import { VisuallyHidden } from '@react-aria/visually-hidden';
+import { useToggleState } from '@react-stately/toggle';
+import { useFocusRing } from '@react-aria/focus';
+import React from 'react';
+
+interface SwitchProps extends AriaSwitchProps {
+  isSelect: boolean;
+}
+
+export default function Switch(props: SwitchProps) {
+  let state = useToggleState({ ...props, isSelected: props.isSelect });
+  let ref = React.useRef(null);
+  let { inputProps } = useSwitch(props, state, ref);
+  let { isFocusVisible, focusProps } = useFocusRing();
+
+  console.log(state);
+
+  return (
+    <label
+      style={{
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'space-between',
+        opacity: props.isDisabled ? 0.4 : 1
```

Comment:
add a disable property for the input instead of just setting the opacity as the input state will be changed anyway